src/main/java/com/clouway/cuse/spi/SearchEngineImpl.java

```diff
@@ -38,6 +38,16 @@ public void register(Object instance) {
 
   }
 
+  public void registerAll(List<Object> objects){
+    Class instanceClass=objects.get(0).getClass();
```

Comment:
Checking for the list size is added. If an empty list is passed an EmptyIndexList exception is thrown.